src/ipc/ipc-zephyr.c

```diff
@@ -328,6 +328,9 @@ int ipc_platform_send_msg(const struct ipc_msg *msg)
 	/* prepare the message and copy to mailbox */
 	struct ipc_cmd_hdr *hdr = ipc_prepare_to_send(msg);
 
+	if (msg->tx_size)
```

Comment:
Should ```mailbox_dspbox_write()``` check this instead (meaning we only check in one place) ?

Reply:
@lgirdwood  Ack. If no blockers, I'll proceed to move the dspbox writes to the IPC driver in Zephyr, so that use of mailbox is abstracted in SOF side (SOF ipc-zephyr.c just uses the driver interface and has no direct use of mailbox.h). The "if (msg->tx_size)" check can be then done in the IPC backend driver in Zephyr.